azext_iot/operations/_mqtt.py

```diff
@@ -5,92 +5,70 @@
-
-        self.client = mqtt.Client(protocol=mqtt.MQTTv311, client_id=device_id)
-        self.client.on_connect = self.on_connect
-        self.client.on_message = self.on_message
-        self.client.on_publish = self.on_publish
-        self.client.tls_set(ca_certs=tls["ca_certs"], tls_version=tls["tls_version"])
-        self.client.username_pw_set(username=build_mqtt_device_username(target["entity"], device_id), password=sas)
-        self.client.connect(host=self.target["entity"], port=8883)
-
-    def on_connect(self, client, userdata, flags, rc):
-        six.print_(
-            "Connected to target IoT Hub MQTT broker with result: {}".format(
-                connection_result[rc]
-            )
-        )
-        self.client.subscribe(self.topic_receive)
-        six.print_("Subscribed to device bound message queue")
-        self.connected = True
-
-    def on_message(self, client, userdata, msg):
+        self.device_client = mqtt_device_client.create_from_connection_string(device_conn_string)
+        self.device_client.connect()
+        self.device_client.on_message_received = self.message_handler
+        self.device_client.on_method_request_received = self.method_request_handler
+        self.method_response_code = method_response_code
+        self.method_response_payload = method_response_payload
+
+    def send_d2c_message(self, message_text="", properties={}):
+        message = Message(message_text)
+        message.custom_properties = properties
+        self.device_client.send_message(message)
+
+    def message_handler(self, message):
         six.print_()
-        six.print_("_Received C2D message with topic_: {}".format(msg.topic))
-        six.print_("_Payload_: {}".format(msg.payload))
-
-    def on_publish(self, client, userdata, mid):
-        six.print_(".", end="", flush=True)
+        six.print_("_Received C2D message with topic_: /devices/{}/messages/devicebound".format(self.device_id))
+        six.print_("_Payload_: {}".format(message.data))
+        if message.custom_properties:
+            six.print_("_Custom Properties_: {}".format(message.custom_properties))
 
-    def is_connected(self):
-        return self.connected
-
-    def execute(self, data, publish_delay=2, msg_count=100):
+    def method_request_handler(self, method_request):
+        six.print_()
+        six.print_("Received method request with id: '{}' and method name: '{}' for device with id: '{}'".format(
+            method_request.request_id, method_request.name, self.device_id))
+        six.print_("_Payload_: {}".format(method_request.payload))
+
+        # set response payload
+        if self.method_response_payload:
+            payload = self.method_response_payload
+        else:
+            payload = {
+                "device_id": self.device_id,
+                "method_name": method_request.name,
+                "method_request_id": method_request.request_id,
+                "data": "Method executed successfully"
```

Comment:
If no custom payload is requested, the returned payload needs to show the request payload

Reply:
Also this needs to be formatted with json casing

Reply:
Example:

```
{
  "methodRequestName": "mymethod",
  "methodRequestId": 12345,
  "methodRequestPayload": { ... }
}
```

I didn't include device Id because I don't think it is that useful. If a target device does not have a method callback setup the `invoke-device-method` command will fail or timeout (which includes the target device you are making a request to anyway).